evalml/model_understanding/_partial_dependence_utils.py

```diff
@@ -221,45 +225,105 @@ def _cartesian(arrays):
     Args:
         pipeline (PipelineBase): pipeline.
         grid (pd.DataFrame): Grid of features to compute the partial dependence on.
         features (list(str)): Column names of input data
         X (pd.DataFrame): Input data.
+        X_training (pd.DataFrame, np.ndarray): The data that was used to train the original pipeline. Will
+            be used in fast mode to train the cloned pipelines.
+        y_training (pd.Series, np.ndarray): The target data that was used to train the original pipeline. Will
+            be used in fast mode to train the cloned pipelines.
+        fast_mode (bool, optional): Whether or not performance optimizations should be
+            used. Defaults to False. When True, copies of pipelines will be used to transform just the
+            column(s) we're calculating partial dependence for. This means that any pipeline
+            containing a component that relies on multiple columns for fit and transform should
+            not be used. See the ``_can_be_used_for_fast_partial_dependence`` property on components
+            to determine which components cannot be used for fast mode.
 
     Returns:
         Tuple (np.ndarray, np.ndarray): averaged and individual predictions for
             all points in the grid.
     """
-    predictions = []
-    averaged_predictions = []
+    X_eval = X.ww.copy()
+    prediction_object = pipeline
+    if fast_mode:
+        # In fast mode, we alter the transformed X with our partial dependence values
+        # and then call predict directly with the estimator
+        X_eval = pipeline.transform_all_but_final(X_eval)
+        prediction_object = pipeline.estimator
+
+        # Some components may drop features, so we need to know whether the specified
+        # features actually have an impact on predictions
+        feature_provenance = pipeline._get_feature_provenance()
+        variable_has_features_passed_to_estimator = {
+            variable: variable in feature_provenance or variable in X_eval.columns
+            for variable in features
+        }
+        no_features_passed_to_estimator = not any(
+            variable_has_features_passed_to_estimator.values(),
+        )
+
+        # Fit pipelines for each feature so that we can transform it with grid
+        # values later
+        cloned_feature_pipelines = _get_cloned_feature_pipelines(
+            features,
+            pipeline,
+            variable_has_features_passed_to_estimator,
+            X_training,
+            y_training,
+        )
 
     if is_regression(pipeline.problem_type):
-        prediction_method = pipeline.predict
+        prediction_method = prediction_object.predict
     else:
-        prediction_method = pipeline.predict_proba
+        prediction_method = prediction_object.predict_proba
+    if fast_mode and no_features_passed_to_estimator:
+        original_predictions = prediction_method(X_eval)
+        original_predictions_mean = np.mean(original_predictions, axis=0)
```

Comment:
In theory we could do this for slow mode too-- it would require us to calculate `pipeline.transform_all_but_final(X_eval)`, though, and I didn't do a full set of analysis on whether that ends up saving time in many cases, so I decided to not add it in for now. I created [an issue](https://github.com/alteryx/evalml/issues/3807) to look into adding it at a separate time.